update TiDB roadmap for CY24-CY25 (#17379)
pingcap/docs: tidb-roadmap.md

Reviewer: Preview: https://pingcap-docsite-preview-git-preview-pi-3d0089-oreoxmts-projects.vercel.app/tidb/dev/tidb-roadmap

      <td>
        <ul>
          <li>
            <b>Limit memory consumption for backup tasks</b><br />
          </li>
          <br />
          <li>
            <b>Limit memory consumption for statistics collection</b><br />
          </li>
          <br />
          <li>
            <b>Manage massive SQL bindings</b><br />
              Improve the user experience of SQL binding, making it convenient for users to create and manage a large number of execution plans to stabilize database performance.
          </li>
          <br />
          <li>
            <b>Enhance resource group control over complex SQL</b><br />
              Regularly assess the Request Unit (RU) consumption of complex SQL before completion of SQL execution to prevent excessively large impacts on the entire system during execution.
          </li>
          <br />
          <li>
            <b>Automatically switch resource groups for runaway queries</b><br />

>               When a query is identified as a runaway query, users can choose to adjust it to a specific resource group and set an upper limit on resource consumption.

Author: ```suggestion
              When a query is identified as a runaway query, users can choose to switch it to a specific resource group and set an upper limit on resource consumption.
```

          <br />
          <li>
            <b>Limited memory for stats collections</b><br />
          </li>
          <br />
          <li>
            <b>Managing massive SQL bindings</b><br />
              Improve the user experience of importing and managing a large number of SQL bindings. 
          </li>
          <br />
          <li>
            <b>Charging RU intermittently for an expensive SQL</b><br />
              Measure the RU produced before one large query completes, which mitigates the impacted by expensives SQL in resource groups.
          </li>
          <br />
          <li>
            <b>Switching resource groups of runaway queries</b><br />
              When a query is identified as runaway queries, users can put it to a particular resource group, in order to limit its resource. 
          </li>
        </ul>
      </td>
      <td>
        <ul>
          <li>
            <b>Limited memory for schema meta</b><br />
          </li>
          <br />
          <li>
            <b>Distributed statistic collection</b><br />
            Collecting statistics in parallel across multiple TiDB instances.
          </li>
          <br />
          <li>

>             <b>Multi-versioned statistics多版本统计信息</b><br />

Reviewer 2: ```suggestion
            <b>Multi-versioned statistics</b><br />
```

      </td>
      <td>
        <ul>
           <li>
            <b>Reliable query termination</b><br />
            Running SQL statements can be immediately terminated, and the corresponding resources are released from TiDB and TiKV.
          </li>
          <br />
          <li>
            <b>Permission control for switching resource groups</b>
            <br />Only users with specific permissions can switch their resource groups, thus preventing resource abuse.
          </li>
          <br />
          <li>
            <b>Mapping tables or SQL with hot Regions</b>
          </li>
          <br />
          <li>
            <b>Logical data import mode with <code>IMPORT INTO</code></b>
          </li>
          <br />
        </ul>
      </td>
      <td>
        <ul>
          <li>
            <b>Fine-grained customization of statistics collection</b>
            <br />Users can modify the statistics collection strategy for specific tables, such as healthiness and parallelism.
          </li>
          <br />
          <li>
            <b>Workload Repository</b>

>             <br />TiDB persists workload information in memory, including cumulative and real-time statistic data, which aids in troubleshooting and analysis.

Author: ```suggestion
            <br />TiDB persists workload information in memory to permanent volume, including cumulative and real-time statistic data, which aids in troubleshooting and analysis.
```

          </li>
          <br />
          <li>
            <b>Automatic index advisor</b>
            <br />TiDB automatically analyzes SQL statements that can be optimized and recommends creating or dropping indexes.
          </li>
          <br />
          <li>
            <b>Support modifying column types for partitioned tables</b>
            <br />Users can modify the data type of columns in partitioned tables, regardless of whether a column is a partitioning key.
          </li>
          <br />
          <li>
            <b>Conflict strategy for <code>IMPORT INTO</code></b>
            <br />Users can set the conflict resolution strategy when importing data, such as exiting with an error, ignoring, or replacing in case of conflicts.
          </li>
          <br />
          <li>
            <b>End-to-End monitoring</b>
            <br />Track the time consumption of individual SQL statements throughout their entire lifecycle, including consumption on TiDB, TiKV, PD, and TiFlash components.
          </li>
        </ul>
      </td>
      <td>
        <ul>
          <li>
            <b>Workload analysis</b>

>             <br />Analyze historical workload data from the Workload Repository and provide optimization recommendations, such as SQL tuning and adjustments to statistics collection strategies.

Author: ```suggestion
            <br />Analyze historical workload data from the Workload Repository and provide optimization recommendations, such as SQL tuning and statistics collection.
```

      <td>
        <ul>
          <li>
            <b>Google Cloud KMS</b>
            <br />Enhance the key management mechanism for static encryption based on Google Cloud KMS, making it generally available.
          </li>
          <br />
          <li>
            <b>Improved dynamic privilege</b>
            <br />Improve the dynamic privilege design and limit the implementation of Super privilege.
          </li>
          <br />
          <li>
            <b>Marker-based log desensitization</b>

>             <br />Support marking sensitive information in the cluster log, and then you can determine whether to desensitize it according to the usage scenario.

Reviewer 3: ```suggestion
            <br />Support marking sensitive information in the cluster log. Then, you can determine whether to desensitize it according to the usage scenario.
```

          </li>
          <br />
          <li>
            <b>FIPS</b>
            <br />Encryption scenarios comply with FIPS.
          </li>
        </ul>
      </td>
      <td>
        <ul>
           <li>
            <b>IAM authentication for AWS</b>
            <br />TiDB as AWS third-party ARN for AWS IAM access.
          </li>
          <br />
          <li>
            <b>Kerberos</b>
            <br />Support Kerberos-based authentication.
          </li>
          <li>
            <b>MFA</b>
            <br />Support multi-factor authentication mechanism.
          </li>
        </ul>
      </td>
      <td>
        <ul>
          <li>
            <b>Label-based access control</b>
            <br />Support data access control by configuring labels.
          </li>
          <br />
          <li>
            <b>Enhanced client-side encryption</b>
            <br />Support client-side encryption of key fields to enhance data security.
          </li>
          <br />
          <li>
            <b>Dynamic desensitization of business data</b>

>             <br />Data desensitization based on different data application scenarios to ensure data security in important fields.

Reviewer 3: ```suggestion
            <br />Support desensitizing data based on different data application scenarios to ensure data security in important fields.
```
